java-util/src/main/java/io/druid/java/util/common/parsers/ParserUtils.java

```diff
@@ -95,4 +99,15 @@ public static String stripQuotes(String input)
     }
     return input;
   }
+
+  /**
+   * Return a function to generate default column names.
+   * Note that the postfix for default column names starts from 1.
+   *
+   * @return column name generating function
+   */
+  public static IntFunction<String> getDefaultColumnNameGenerator()
```

Comment:
IMO it's overengineering to to prepare for different "strategies" of column name generation, it could be just a static method accepting int. generateFieldNames() and abstract parser use this method.